lab_6_pipeline/pipeline.py

```diff
@@ -18,21 +49,56 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self._storage = {}
+        self.path = path_to_raw_txt_data
+        self._validate_dataset()
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not isinstance(self.path, Path) or not self.path.exists():
+            raise FileNotFoundError
+
+        if not self.path.is_dir():
+            raise NotADirectoryError
+
+        if not self.path.glob("*"):
+            raise EmptyDirectoryError
+
+        meta_list = [elem.name for elem in self.path.glob("*.json")]
+        raw_list = [elem.name for elem in self.path.glob("*.txt")]
+        number_list = sorted([int(re.search(r'\d+', elem)[0]) for elem in meta_list])
+        right_list = [i for i in range(1, len(number_list)+1)]
```

Comment:
cast to list instead of comprehension

Reply:
do you mean like this: `right_list = list(range(1, max(number_list)+1))`?